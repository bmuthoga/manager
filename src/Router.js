import React, { Component } from 'react';
import { Scene, Router } from 'react-native-router-flux';
import { connect } from 'react-redux';
import LoginForm from './components/LoginForm';
import EmployeeList from './components/EmployeeList';
import EmployeeCreate from './components/EmployeeCreate';
import EmployeeEdit from './components/EmployeeEdit';
import { openEmployeeCreate } from './actions';

class RouterComponent extends Component {
  render() {
    return (
      // Can add prop sceneStyle={{ paddingTop: 65 }} to below Router tag if scene title is
      // overlaying other elements
      <Router>
        <Scene key="root">
          <Scene key="auth" hideNavBar>
            <Scene
              key="login"
              component={LoginForm}
              title="Please Login"
              hideNavBar={false}
            />
          </Scene>

          <Scene key="main" hideNavBar>
            <Scene
              key="employeeList"
              component={EmployeeList}
              title="Employees"
              renderBackButton={() => null}
              hideNavBar={false}
              onRight={() => this.props.openEmployeeCreate()}
              rightTitle="Add"
            />
            <Scene
              key="employeeCreate"
              component={EmployeeCreate}
              title="Create Employee"
              hideNavBar={false}
            />
            <Scene
              key="employeeEdit"
              component={EmployeeEdit}
              title="Edit Employee"
              hideNavBar={false}
            />
          </Scene>
        </Scene>
      </Router>
    );
  }
}

export default connect(null, { openEmployeeCreate })(RouterComponent);
