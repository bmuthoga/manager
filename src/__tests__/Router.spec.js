import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import Router from '../Router';

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {};

describe('>>>R O U T E R --- Test Router', () => {
    it('+++ renders correctly', () => {
        const wrapper = shallow(
            <Router />,
            { context: { store: mockStore(initialState) } }
        );

        expect(wrapper.dive()).toMatchSnapshot();
    });
});
