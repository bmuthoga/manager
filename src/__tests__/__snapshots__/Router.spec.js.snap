// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>R O U T E R --- Test Router +++ renders correctly 1`] = `
<Router>
  <Component
    key="root"
  >
    <Component
      hideNavBar={true}
      key="auth"
    >
      <Component
        component={[Function]}
        hideNavBar={false}
        key="login"
        title="Please Login"
      />
    </Component>
    <Component
      hideNavBar={true}
      key="main"
    >
      <Component
        component={[Function]}
        hideNavBar={false}
        key="employeeList"
        onRight={[Function]}
        renderBackButton={[Function]}
        rightTitle="Add"
        title="Employees"
      />
      <Component
        component={[Function]}
        hideNavBar={false}
        key="employeeCreate"
        title="Create Employee"
      />
      <Component
        component={[Function]}
        hideNavBar={false}
        key="employeeEdit"
        title="Edit Employee"
      />
    </Component>
  </Component>
</Router>
`;
